File: SNS.Web/src/app/discussions/problems/problems/contracts/problem-details.dto.ts
import { CommunitySnapshotDto } from '../../../../shared/contracts/community-snapshot.dto';
import { ProfileSnapshotDto } from '../../../../profiles/profiles/contracts/profile-snapshot.dto';
import { DifficultyLevel } from '../../../shared/enums/difficulty-level.enum';
import { VoteType } from '../../../shared/enums/vote-type.enum';
import { ProblemBlockType } from '../../enums/problem-block-type.enum';
import { ProblemStatus } from '../../enums/problem-status.enum';
import { ProblemContentBlockDto } from './problem-content-block.dto';

export interface ProblemDetailsDto {
    id: string;
    title: string;
    status: ProblemStatus;
    level: DifficultyLevel;
    createdAt: string;
    updatedAt: string;
    author: ProfileSnapshotDto;
    community: CommunitySnapshotDto | null;
    contentBlocks: ProblemContentBlockDto[];
    tags: string[];
    topics: string[];
    upvotesCount: number;
    downvotesCount: number;
    solutionsCount: number;
    viewsCount: number;
    currentUserVote: VoteType | null;
}

File: backend/SNS.Application/Discussions/Problems/Problems/Contracts/ProblemDetailsDto.cs
using SNS.Application.ContentManagement.Communities.Communities.Contracts;
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Domain.Discussions.Problems.Enums;
using SNS.Domain.Discussions.Shared.Enums;

namespace SNS.Application.Discussions.Problems.Problems.Contracts;

/// <summary>
/// Represents full details of a discussion problem including author, community, content blocks, metrics, and relationship state.
/// </summary>
/// <param name="Id">The unique identifier of the problem.</param>
/// <param name="Title">The problem title.</param>
/// <param name="Status">The lifecycle status.</param>
/// <param name="Level">The difficulty level.</param>
/// <param name="CreatedAt">The creation timestamp.</param>
/// <param name="UpdatedAt">The last update timestamp.</param>
/// <param name="Author">Snapshot overview of the problem author.</param>
/// <param name="Community">Optional snapshot overview of the community.</param>
/// <param name="ContentBlocks">The ordered list of content blocks.</param>
/// <param name="Tags">The list of associated tags.</param>
/// <param name="Topics">The list of AI-classified topics.</param>
/// <param name="UpvotesCount">The count of upvotes.</param>
/// <param name="DownvotesCount">The count of downvotes.</param>
/// <param name="SolutionsCount">The count of solutions.</param>
/// <param name="ViewsCount">The total views recorded.</param>
/// <param name="CurrentUserVote">The vote type cast by the requesting user, if any.</param>
public sealed record ProblemDetailsDto(
    Guid Id,
    string Title,
    ProblemStatus Status,
    DifficultyLevel Level,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    ProfileSnapshotDto Author,
    CommunitySnapshotDto? Community,
    List<ProblemContentBlockDto> ContentBlocks,
    List<string> Tags,
    List<string> Topics,
    int UpvotesCount,
    int DownvotesCount,
    int SolutionsCount,
    int ViewsCount,
    VoteType? CurrentUserVote
);